flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/internal/JdbcOutputFormat.java

```diff
@@ -154,7 +154,9 @@ public void open(int taskNumber, int numTasks) throws IOException {
                                         try {
                                             flush();
                                         } catch (Exception e) {
-                                            flushException = e;
+                                            if (flushException == null) {
+                                                flushException = e;
+                                            }
```

Comment:
`flush()` may be called multiple times by the timer before the main thread discovers the exception and there might actually be multiple exceptions. To let users see them all, consider maintaining a list. Make the first exception as the cause and the other exceptions as suppressed by calling `Throwable#addSuppressed`.

Reply:
Thanks for reply! 
When the first time `flush()` cause an exception, the `flushException` is set. And the next time `flush()` is called, it will only throw `RuntimeException("Writing records to JDBC failed.", flushException)` without execute real flush logic.
So, the only useful information is recorded in the first exception stack. And I think there is no need to record the following exceptions.